content/en/blog/2024/inpod-traffic-redirection-ambient/index.md

```diff
@@ -0,0 +1,228 @@
+publishdate: 2024-01-26
+attribution: "Ben Leggett (Solo.io), Yuval Kohavi (Solo.io), Lin Sun (Solo.io)"
+keywords: [Ambient,Istio,CNI,ztunnel,traffic]
+---
+
+The Istio project [announced its innovative new ambient mode](/blog/2022/introducing-ambient-mesh/) in 2022, and [released an Alpha implementation](/news/releases/1.18.x/announcing-1.18/#ambient-mesh) in early 2023.
+
+Our Alpha was focused on proving out the value of the ambient model, under limited conditions, which it certainly has done. However, the conditions were quite limited. Ambient mode relies on transparently redirecting traffic, and the initial mechanism we used to do that conflicted with many CNI and networking implementations.
+Through GitHub issues and Slack discussions, we heard our users wanted to be able to use ambient mode in [minikube](https://github.com/istio/istio/issues/46163) and [Docker Desktop](https://github.com/istio/istio/issues/47436), with CNI implementations like [Cilium](https://github.com/istio/istio/issues/44198) and [Calico](https://github.com/istio/istio/issues/40973), and on services like [OpenShift](https://github.com/istio/istio/issues/42341) and [Amazon EKS](https://github.com/istio/istio/issues/42340).
+Getting broad support for Kubernetes anywhere has become the No. 1 requirement for ambient mesh moving to Beta — people have come to expect Istio to work on any Kubernetes platform and with any Container Network Interface (CNI) layer. After all, Ambient wouldn’t be ambient without being all around you!
+
+At Solo, we've been integrating ambient mode into our Gloo Mesh product, and came up with an innovative solution to this problem. We had decided
+to [upstream](https://github.com/istio/istio/issues/48212) our changes in late 2023 to help Ambient reach beta faster
+so more users can operate ambient in Istio 1.21 or newer and enjoy the benefits of ambient sidecar-less mesh in their
+platforms regardless of their preferred CNIs.
+
+## How did we get here?
+
+### Service meshes and CNIs: it's complicated
+
+Istio is a service mesh, and service meshes by definition are not *CNI implementations* - service meshes build on top of the
+low-level, standards-compliant CNI implementation present in Kubernetes clusters to offer the high-level policy and security
+controls users need at scale.
+
+Basically, before you can do things like secure pod traffic with mTLS and apply lots of high-level authentication and authorization policy
+at the service mesh layer, you must have a functional Kubernetes cluster with a functional CNI implementation, to make sure the
+basic networking pathways are set up so that packets can get from one pod to another (and from one node to another) in
+your cluster.
+
+Though some service meshes may also ship and require their own in-house CNI implementation, the Istio project has chosen not
+to, for maximum compatibility with managed offerings, cross-vendor support, and composability with the broader CNCF ecosystem.
```

Comment:
```suggestion
Though some service meshes may also ship and require their own in-house primary CNI implementation, and it is sometimes possible to run two primary CNI implementations in parallel within the same cluster, in practice this introduces a whole host of compatibility issues, strange behaviors, reduced featuresets, and simple incompatibilities due to the wildly varying mechanisms each CNI implementation might employ internally.

To avoid this, the Istio project has chosen not to ship or require their own primary CNI implementation, or even require a "preferred" CNI implementation - instead choosing to support CNI chaining with the widest possible ecosystem of CNI implementations, and ensuring maximum compatibility with managed offerings, cross-vendor support, and composability with the broader CNCF ecosystem.
```